### PR TITLE
Always allow reorder for triggers, conditions, actions and options

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Some changes in this PR :
1. **Re-order mode is gone** : reordering action is always available on desktop (width>600px) and disabled on mobile because we don't have many space for the drag handle and sortable component doesn't work well on mobile. Copy, cut, duplicate are still available on mobile.
2. **Move up and move down arrows** : These button are still available in the overflow menu next to copy/cut/duplicate actions to avoid overloading the interface because reordering action is always available.

CleanShot 2024-01-30 at 09 48 23

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/automation/action/types/ha-automation-action-choose.ts [hunk 1]
```diff
@@ -64,12 +65,31 @@ export class HaChooseAction extends LitElement implements ActionElement {
...
 
   private _expandLast = false;
 
+  private _mqlListenerRef?: (event: MediaQueryListEvent) => void;
+
+  private _mql?: MediaQueryList;
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this._mql = window.matchMedia("(min-width: 600px)");
```
Comment (bramkragten) [inline src/panels/config/automation/action/types/ha-automation-action-choose.ts:78]: Does this have to be `600px`, can we maybe use `narrow`?

https://github.com/home-assistant/frontend/blob/dev/src/layouts/home-assistant-main.ts#L54

File: src/panels/config/automation/action/ha-automation-action.ts [hunk 2]
```diff
@@ -53,6 +53,27 @@ export default class HaAutomationAction extends LitElement {
...
+  private _mql?: MediaQueryList;
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this._mql = window.matchMedia("(min-width: 600px)");
+    this._showReorder = this._mql!.matches;
+    this._mqlListenerRef = (event: MediaQueryListEvent) => {
+      this._showReorder = event.matches;
+    };
+    this._mql.addListener(this._mqlListenerRef);
```
Comment (bramkragten) [inline src/panels/config/automation/action/ha-automation-action.ts:67]:
```suggestion
    this._unsubMql = listenMediaQuery("(min-width: 600px)", (event: MediaQueryListEvent) => {
      this._showReorder = event.matches;
    });
```

File: src/panels/config/automation/action/ha-automation-action.ts [hunk 3]
```diff
@@ -53,6 +53,27 @@ export default class HaAutomationAction extends LitElement {
...
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this._mql?.removeListener(this._mqlListenerRef!);
+    this._mqlListenerRef = undefined;
+    this._mql = undefined;
```
Comment (bramkragten) [inline src/panels/config/automation/action/ha-automation-action.ts:74]:
```suggestion
    this._unsubMql?.();
    this._unsubMql = undefined;
```

File: src/panels/config/automation/action/types/ha-automation-action-choose.ts (hunk 1 above)
Comment (piitaya) [reply src/panels/config/automation/action/types/ha-automation-action-choose.ts:78]: Narrow is bigger (870px) and we have to pass it to every component so I preferred to use it like this.

File: src/panels/config/automation/action/ha-automation-action.ts (hunk 2 above)
Comment (piitaya) [reply src/panels/config/automation/action/ha-automation-action.ts:67]: Thanks you! I can not found this 🙂

**Review by bramkragten — APPROVED** [review]

**Pull request merged**